spine/events.go

```diff
@@ -87,7 +87,7 @@ func (r *events) Publish(payload api.EventPayload) {
 	}
 
 	for _, level := range handlerLevels {
-		for _, item := range r.handlers {
+		for _, item := range handler {
```

Comment:
Why do you interate over the structure variable instead of the local variable? If Publish is invoked again, wouldn't this trigger a race condition?